src/sa/internal/sa_interface_inmemory.template.c

```diff
@@ -1510,12 +1521,17 @@ static int32_t sa_setARSNW(void)
 
         for (x = 0; x < sa[spi].arsnw_len; x++)
         {
-            sa[spi].arsnw = (((uint8_t)sdls_frame.pdu.data[x + 3]) << (sa[spi].arsnw_len - x));
+            sa[spi].arsnw = (((uint8_t)sdls_frame.pdu.data[x + 2]));
```

Comment:
I'm not sure this makes sense as a for loop as it continually will overwrite the arsnw

Reply:
Correct, I fixed the shift but forgot to take the loop out